include/zephyr/bluetooth/bluetooth.h

```diff
@@ -685,6 +685,26 @@ enum {
 	 * @note Requires @ref BT_LE_ADV_OPT_USE_NAME
 	 */
 	BT_LE_ADV_OPT_FORCE_NAME_IN_AD = BIT(18),
+
+	/**
+	 * @brief Use the Non-Resolvable Private Address when advertise as
+	 * connectable.
+	 *
+	 * By default, when starting advertising using @ref
+	 * BT_LE_ADV_OPT_CONNECTABLE and @kconfig{CONFIG_BT_PRIVACY} is
+	 * disabled, the advertiser set uses an identity address as the
+	 * advertising address.
+	 * When this option is enabled together with @ref
+	 * BT_LE_ADV_OPT_CONNECTABLE, the advertiser set will use freshly
+	 * generated Non-Resolvable Private Address on every advertising start.
+	 * If this option gets disabled, the advertiser set starts using the
+	 * identity address back again.
+	 *
+	 * @note Requires @ref BT_LE_ADV_OPT_CONNECTABLE.
+	 *
+	 * @note Has no effect if @kconfig{CONFIG_BT_PRIVACY} is enabled.
```

Comment:
Can we simplify the semantics on this and make it have the same effect independent of what this kconfig is set to?

Reply:
It doesn't seem to be possible as the behavior with `BT_PRIVACY` in general is different. But as long as behavior is clarified in the description, the current naming should be fine.

This option is the opposite to `BT_LE_ADV_OPT_USE_IDENTITY`. But, when `BT_PRIVACY` is disabled, the host for some reasons (which aren't clear to me) restores the identity address if the address is random, instead of regenerating the new random address. If the host didn't do this, this new option wouldn't be needed as I could get the current behavior with setting `BT_LE_ADV_OPT_USE_IDENTITY` and keeping the identity address unchanged, and get the desired behavior with removing the `BT_LE_ADV_OPT_USE_IDENTITY` option.

(I rephrased this description a bit, renamed the option and added params check for it when it 
 is set with `BT_LE_ADV_OPT_USE_IDENTITY`).

Reply:
> which aren't clear to me

I think we should try to make sense of this before making any changes to the behavior.

Reply:
> I think we should try to make sense of this before making any changes to the behavior.

That is question to code owners I believe :)

Reply:
> > which aren't clear to me
> 
> I think we should try to make sense of this before making any changes to the behavior.

Seems like there is no particular reason for not updating address on every adv start according to this discussion: https://github.com/zephyrproject-rtos/zephyr/pull/27907#discussion_r480180674